File: dotnet/Platforms/GameX.Platform.Ex/Ex_Render.cs
using OpenStack.Gfx;
using OpenStack.Gfx.Ex;
#pragma warning disable CS9113

namespace GameX.Platforms.Ex;

public static class ExRenderer {
    public static Renderer CreateRenderer(object parent, IOpenGfx[] gfx, object obj, string type) {
        if (obj is IHaveOpenGfx z) gfx = z.Gfx;
        return type switch {
            "TestTri" => new TestTriRenderer(gfx, obj),
            "Texture" => new SpriteRenderer(gfx, obj),
            //"Object" => new ObjectRenderer(gfx, obj),
            _ => default
        };
    }
}

File: python/gamex/platforms/ex/ex_render.py
from __future__ import annotations
import os
from openstk.gfx import IHaveOpenGfx
from openstk.platforms.ex.gfx import TestTriRenderer

# typedefs
class IOpenGfx: pass
class Renderer: pass

# ExRenderer
class ExRenderer:
    @staticmethod
    def createRenderer(parent: object, gfx: list[IOpenGfx], obj: object, type: str) -> Renderer:
        if isinstance(obj, IHaveOpenGfx): gfx = obj.gfx
        surf = parent.surface
        match type:
            case 'TestTri': return TestTriRenderer(gfx, obj, surf)
            case _: return None
